### PR TITLE
cli: consume external runtime-mappings

### PR DESCRIPTION
**Pull request by lucab**
This updates tectonic-torcx to consume runtime-mappings coming from an
external source. Mapping file will be bind-mounted at runtime:
 * boostrapper will receive the one provided by the installer
 * pre-hook will receive it as a config-map from the cluster

### CODE REVIEW

**Comment by lucab** [comment]
/cc @squeed 
This currently still misses one change in the bootstrapper in order to directly query the api-server to check/retrieve the config-map, before falling back on the installer file.

File: cli/common.go [hunk 1]
```diff
@@ -63,12 +63,12 @@ func commonFlags(f *pflag.FlagSet) {
...
 	f.BoolVar(&cfg.NoVerifySig, "no-verify-signatures", false, "don't gpg-verify all downloaded addons")
 	f.StringVar(&cfg.GpgKeyringPath, "keyring", "/pubring.gpg", "path to the gpg keyring")
-	f.StringVar(&cfg.VersionManifestPath, "version-manifest", "/versions.yaml", "path to the version manifest file")
+	f.StringVar(&cfg.VersionManifestPath, "version-manifest", "", "path to the runtime-mappings manifest file")
 	f.StringVar(&verbose, "verbose", "info", "verbosity level")
 }
 
 // parseFlags parses CLI options, returning a populated configuration for the bootstrap agent
-func parseFlags() (internal.Config, error) {
+func parseFlags(runtimeMappingsPath string) (internal.Config, error) {
```
Comment (squeed) [inline cli/common.go]: can you call this something like "defaultRuntimeMappingsPath"? Just so it's clear that this is a fallback.

File: internal/versions.go [hunk 2]
```diff
@@ -81,19 +100,72 @@ func parseVersionManifest(data []byte) (*VersionManifest, error) {
...
+func (a *App) GetVersionManifest(localOnly bool) (*VersionManifest, error) {
+	path := a.Conf.VersionManifestPath
+	if path == "" {
+		return nil, errors.New("missing version manifest path")
+	}
+
+	// Conditionally try ConfigMap from api-server first (bootstrapper only)
+	if !localOnly {
+		logrus.Debug("Querying api-server for runtime mappings ConfigMap")
+		manifest, err := a.versionFromAPIServer()
```
Comment (squeed) [inline internal/versions.go]: should this be versionManifestFromAPIServer?

**Comment by squeed** [comment]
Seems reasonable. I somehow wish there were a simpler way to plumb this all up, but I can't think of one.

File: internal/versions.go (hunk 2 above)
Comment (lucab) [reply internal/versions.go]: Definitely a good catch! It looks like I fixed this locally but forgot to push :(

File: cli/common.go (hunk 1 above)
Comment (lucab) [reply cli/common.go]: Ack. I've also left a docstring note to describe argument usage.

**Comment by lucab** [comment]
@squeed this is ready to go, PTAL. We can tag `v0.2.0` after this and switch tectonic-installer to it. `installer-latest` tag will proceed in the `image-installer-latest` branch, which we should document.

**Comment by squeed** [comment]
You should update the example hook yaml file to include the configmap.

**Comment by lucab** [comment]
@squeed yep thanks, pushed an additional commit with deployment examples.

**Pull request merged**